runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -9754,21 +9574,35 @@ J9::X86::TreeEvaluator::vectorizedHashCodeHelper(TR::Node *node, TR::DataType dt
    else if (comp->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX2))
       vl = TR::VectorLength256;
 
-   TR::Register *address = cg->evaluate(node->getChild(0));
+   TR::Node *addressNode = node->getChild(0);
+
+   bool nonZeroOffset = node->getChild(1)->getOpCodeValue() != TR::iconst || node->getChild(1)->getInt() != 0;
+   bool addressIs64bits = TR::TreeEvaluator::getNodeIs64Bit(addressNode, cg);
+
+   TR::Register *address = nonZeroOffset ? TR::TreeEvaluator::intOrLongClobberEvaluate(addressNode, addressIs64bits, cg) : cg->evaluate(addressNode);
+   TR::Register *offset = cg->evaluate(node->getChild(1));
```

Comment:
Child(1) doesn't need to be evaluated into a register if it is a constant zero since that register isn't used anywhere below when `nonZeroOffset` is true (unless I missed it).  This will just insert an XOR into the code that isn't needed.

If constant 0, you can treat this child just like we do for immediate-form instructions and simply decrement the reference count on the node without evaluating it.